homeassistant/components/lyric/climate.py

```diff
@@ -0,0 +1,281 @@
+        ):
+            self._hvac_modes.append(HVAC_MODE_HEAT_COOL)
+
+        super().__init__(
+            coordinator,
+            location,
+            device,
+            f"{device.macID}_thermostat",
+            device.name,
+            None,
+        )
+
+    @property
+    def supported_features(self) -> int:
+        """Return the list of supported features."""
+        return SUPPORT_FLAGS
+
+    @property
+    def temperature_unit(self) -> str:
+        """Return the unit of measurement."""
+        return self._temperature_unit
+
+    @property
+    def current_temperature(self) -> Optional[float]:
+        """Return the current temperature."""
+        return self.device.indoorTemperature
+
+    @property
+    def hvac_mode(self) -> str:
+        """Return the hvac mode."""
+        return HVAC_MODES[self.device.changeableValues.mode]
+
+    @property
+    def hvac_modes(self) -> List[str]:
+        """List of available hvac modes."""
+        return self._hvac_modes
+
+    @property
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature we try to reach."""
+        device: LyricDevice = self.device
+        if not device.hasDualSetpointStatus:
+            return device.changeableValues.heatSetpoint
+
+    @property
+    def target_temperature_low(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        device: LyricDevice = self.device
+        if device.hasDualSetpointStatus:
+            return device.changeableValues.coolSetpoint
+
+    @property
+    def target_temperature_high(self) -> Optional[float]:
+        """Return the upper bound temperature we try to reach."""
+        device: LyricDevice = self.device
+        if device.hasDualSetpointStatus:
+            return device.changeableValues.heatSetpoint
+
+    @property
+    def preset_mode(self) -> Optional[str]:
+        """Return current preset mode."""
+        return self.device.changeableValues.thermostatSetpointStatus
+
+    @property
+    def preset_modes(self) -> Optional[List[str]]:
+        """Return preset modes."""
+        return [
+            PRESET_NO_HOLD,
+            PRESET_HOLD_UNTIL,
+            PRESET_PERMANENT_HOLD,
+            PRESET_TEMPORARY_HOLD,
+            PRESET_VACATION_HOLD,
+        ]
+
+    @property
+    def min_temp(self) -> float:
+        """Identify min_temp in Lyric API or defaults if not available."""
+        device: LyricDevice = self.device
+        return (
+            device.minCoolSetpoint
+            if LYRIC_HVAC_MODE_COOL in device.allowedModes
+            else device.minHeatSetpoint
+        )
+
+    @property
+    def max_temp(self) -> float:
+        """Identify max_temp in Lyric API or defaults if not available."""
+        device: LyricDevice = self.device
+        return (
+            device.maxHeatSetpoint
+            if LYRIC_HVAC_MODE_HEAT in device.allowedModes
+            else device.maxCoolSetpoint
+        )
```

Comment:
If a ternary spans multiple lines, the Home Assistant standard is to convert it to an `if` block